gen/field.go

```diff
@@ -91,8 +91,10 @@ func (f fieldGroupGenerator) Generate(g Generator) error {
 		return err
 	}
 
-	if err := f.Zap(g); err != nil {
-		return err
+	if !checkNoZap(g) {
```

Comment:
Reads a little funky with the double negative, but I see that the semantics are taken from other similar helper functions like `checkReservedIdentifier`.